### PR TITLE
Functions game state

### PR DESCRIPTION
**Pull request by CorayThan**

### CODE REVIEW

File: src/game/Game.tsx [hunk 1]
```diff
@@ -24,53 +23,27 @@ const config: Phaser.Types.Core.GameConfig = {
     },
 }
 
-class Game extends React.Component<Props> {
-    log: object[] = []
-
-    dispatch = (action: any) => {
-        const { state } = this.props
-        this.log.push(buildLogForAction(action, state))
-        exec(action, state)
```
Comment (granttitus) [inline src/game/Game.tsx]: The code in `src/game` is divided into three systems: 1) the game renderer, 2) the game action manager, 3) the action logger. This file, and more specifically this method, is the glue that binds the three together. You can delete lines #35-40 and the `window.game` line if you must, but the `dispatch` method and `this.log` need to stay. 

The `Phaser.Game` instance renders the game based on a provided state. When a user interacts with a UI element, this game instance triggers an event similar to the one below

```
{
  type: "UseCreature",
  cardID: "jacketattack-creature-2",
  playerName: "jacketattack
}
```

The action is sent to our action manager via the `dispatch` method. The action manager mutates the state, which in turn causes the scene to be re-rendered. The action logger maintains a list of all preformed actions and creates a user friendly message (e.g. "jacketattack reaps with Kelifi Dragon").

To integrate with your code we can give `src/game/Game` a reference to `mergeState` and a callback to call when a new action has been preformed.

File: src/game/Game.tsx [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-import { buildLogForAction } from "./ActionLogger"
+import GameScene from "./GameScene"
 
 interface Props {
     state: object
 }
 
-const width = 1024
-const height = window.innerHeight
+const width = window.innerWidth - chatStore.chatWidth
```
Comment (granttitus) [inline src/game/Game.tsx]: We should pass `width` and `height` in as props. With this file _only_ caring about the game element we'll maintain a solid separation of concerns.

Comment (CorayThan) [reply src/game/Game.tsx]: The width and height here is temporary, and not the real correct width and height anyway. Agreed that these will need to be passed in as props.

File: src/game/Game.tsx [hunk 3]
```diff
@@ -27,36 +28,35 @@ const config: Phaser.Types.Core.GameConfig = {
...
 
-    dispatch = (action: any) => {
-        const { state } = this.props
-        this.log.push(buildLogForAction(action, state))
+    dispatch = (action: Action) => {
+        const {state} = this.props
+        const logObj = buildLogForAction(action, state)
+        log.info("Log is " + prettyJson(logObj))
+        if (logObj != null) {
+            this.log.push(logObj)
```
Comment (granttitus) [inline src/game/Game.tsx]: We can delete `this.log`. It was serving the same purpose `actionStore` is now fulfilling.

Comment (granttitus) [reply src/game/Game.tsx]: Deleted

**Pull request merged**